extensions/scarb-doc/src/types.rs

```diff
@@ -244,6 +247,51 @@ impl Module {
         let module_impls = db.module_impls(module_id).unwrap();
         let impls = chain!(module_impls.keys(), module_pubuses.use_impl_defs.iter())
             .filter(|id| should_include_item(*id))
+            .filter(|impl_def_id| {
+                // Hide impls for hidden traits and hidden trait generic args.
+                // Example: `HiddenTrait<*>` or `NotHiddenTrait<HiddenStruct>` (e.g. Drop<HiddenStruct>).
+                let Ok(trait_id) = db.impl_def_trait(**impl_def_id) else {
+                    return true;
+                };
+                let Ok(item_trait) = db.module_trait_by_id(trait_id) else {
+                    return true;
+                };
+                let all_generic_args_are_hidden = db
+                    .impl_def_concrete_trait(**impl_def_id)
+                    .ok()
+                    .map(|concrete_trait_id| {
+                        let args = concrete_trait_id.generic_args(db.upcast());
+                        if args.is_empty() {
+                            return false;
+                        }
+                        args.into_iter()
+                            .filter_map(|arg_id| {
+                                let GenericArgumentId::Type(type_id) = arg_id else {
+                                    return None;
+                                };
+                                let TypeLongId::Concrete(concrete_type_id) =
+                                    type_id.lookup_intern(db)
+                                else {
+                                    return None;
+                                };
+                                let concrete_id: &dyn SemanticQueryAttrs = match &concrete_type_id {
+                                    ConcreteTypeId::Struct(struct_id) => struct_id,
+                                    ConcreteTypeId::Enum(enum_id) => enum_id,
+                                    ConcreteTypeId::Extern(extern_type_id) => extern_type_id,
+                                };
+                                is_doc_hidden_attr_semantic(db, concrete_id).ok()
+                            })
+                            // We still keep impls, if any trait generic argument is not hidden.
```

Comment:
Nit: I would put this comment right above `all_generic_args_are_hidden` definition